frontend/packages/knative-plugin/src/components/add/CatalogEventSourceForm.tsx

```diff
@@ -0,0 +1,132 @@
+import { isDynamicEventSourceKind } from '../../utils/fetch-dynamic-eventsources-utils';
+
+interface OwnProps {
+  namespace: string;
+  eventSourceStatus: EventSourceListData | null;
+  eventSourceMetaDescription?: any;
+}
+
+const CatalogEventSourceForm: React.FC<FormikProps<FormikValues> & OwnProps> = ({
+  errors,
+  values,
+  handleSubmit,
+  handleReset,
+  setFieldValue,
+  status,
+  isSubmitting,
+  dirty,
+  namespace,
+  eventSourceStatus,
+  eventSourceMetaDescription,
+}) => {
+  const { t } = useTranslation();
+  const yamlEditor = <YAMLEditorField name="yamlData" onSave={handleSubmit} />;
+
+  const sanitizeToForm = (newFormData) => {
+    const specData = newFormData.spec;
+    const initialValue = {
+      project: {
+        name: newFormData.metadata.namespace || '',
+        displayName: '',
+        description: '',
+      },
+      application: {
+        initial: '',
+        name: '',
+        selectedKey: '',
+      },
+      name: newFormData.metadata.name,
+      apiVersion: newFormData.apiVersion,
+      sinkType: newFormData.spec?.sink?.ref ? SinkType.Resource : SinkType.Uri,
+      sink: {
+        apiVersion: newFormData.spec?.sink?.ref?.apiVersion,
+        kind: newFormData.spec?.sink?.ref?.kind,
+        name: newFormData.spec?.sink?.ref?.name,
+        key: '',
+        uri: newFormData.spec?.sink?.uri || '',
+      },
+      type: newFormData.kind,
+      data: {
+        [newFormData.kind]: {
+          ..._.omit(specData, 'sink'),
+        },
+      },
+    };
+    return initialValue;
+  };
+
+  const sanitizeToYaml = () =>
+    safeJSToYAML(getCatalogEventSourceResource(values as EventSourceSyncFormData), 'yamlData', {
+      skipInvalid: true,
+    });
+
+  const formEditor = (
+    <>
+      {eventSourceStatus && !_.isEmpty(eventSourceStatus.eventSourceList) && (
+        <div className="row">
+          <div className="col-sm-12 col-md-4 col-md-push-8 col-lg-5 col-lg-push-7">
+            {eventSourceMetaDescription}
+          </div>
+          <div className="col-sm-12 col-md-8 col-md-pull-4 col-lg-7 col-lg-pull-5">
+            {values.showCanUseYAMLMessage && (
+              <Alert
+                actionClose={
+                  <AlertActionCloseButton
+                    onClose={() => setFieldValue('showCanUseYAMLMessage', false)}
+                  />
+                }
+                isInline
+                title={t(
+                  'knative-plugin~Note: Some fields may not be represented in this form view. Please select "YAML view" for full control of object creation.',
+                )}
+                variant="info"
+              />
+            )}
+            <CatalogEventSourceSection namespace={namespace} fullWidth />{' '}
+          </div>
+        </div>
+      )}
+      {eventSourceStatus && !eventSourceStatus.loaded && <LoadingInline />}
+    </>
+  );
+  return (
+    <FlexForm className="odc-pipeline-builder-form__grid" onSubmit={handleSubmit}>
```

Comment:
```suggestion
    <FlexForm onSubmit={handleSubmit}>
```
Not sure, if that is required.